SofaKernel/modules/SofaDefaultType/src/sofa/defaulttype/typeinfo/TypeInfo_Topology.h

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+
+#include <sofa/defaulttype/typeinfo/TypeInfo_FixedArray.h>
+#include <sofa/topology/Topology.h>
+
+namespace sofa::defaulttype
+{
+
+template<>
+struct DataTypeInfo< sofa::topology::geometry::Edge > : public FixedArrayTypeInfo<sofa::type::stdtype::fixed_array<Index,2> >
+{
+    static std::string name() { return "Edge"; }
+};
+
```

Comment:
why do you need this one in addition to the Topology geometry class?

Reply:
This is the Typeinfo-stuff API. so you need that for the Data mechanism.
But I could use the ElementInfo information directly, instead of repeating the same string indeed